docs/man/rpm-dependency-generators.7.scd

```diff
@@ -0,0 +1,406 @@
+RPM-DEPENDENCY-GENERATORS(7)
+
+# NAME
+*rpm-dependency-generators* - RPM file attribute and dependency generator
+interface
+
+# SYNOPSIS
+
+## File attributes
+*%\_\_*_NAME_*_path* _REGEX_++
+*%\_\_*_NAME_*_magic* _REGEX_++
+*%\_\_*_NAME_*_mime* _REGEX_++
+*%\_\_*_NAME_*_exclude_path* _REGEX_++
+*%\_\_*_NAME_*_exclude_magic* _REGEX_++
+*%\_\_*_NAME_*_exclude_mime* _REGEX_
+
+## File attribute tunables
+*%\_\_*_NAME_*_flags* _FLAG_[*,*_FLAG_ ...]++
+*%\_\_*_NAME_*_exclude_flags* _FLAG_[*,*_FLAG_ ...]++
+*%\_local_file_attrs* _NAME_[*:*_NAME_ ...]
+
+_FLAG_ can be one of *exeonly* and *magic_and_path*.
+
+## Generators
+*%\_\_*_NAME_*_*​_TYPE_ _BODY_++
+*%\_\_*_NAME_*_*​_TYPE_*()* _BODY_
+
+_TYPE_ can be one of *conflicts*, *enhances*, *obsoletes*, *orderwithrequires*,
+*provides*, *recommends*, *requires*, *suggests* and *supplements*.
+
+## Generator tunables
+*%\_\_*_NAME_*_protocol* {*simple*|*multifile*}++
+*%\_\_*_NAME_*_*​_TYPE_*_opts* _BODY_++
+*%\_\_*_TYPE_*_exclude* _REGEX_++
+*%\_\_*_TYPE_*_exclude_from* _REGEX_
```

Comment:
Just noticed - the _TYPE_exclude[_from] macros are not generator tunables and MUST NOT be used in generators because they're global for that _TYPE_.

The synopsis subsections don't match what's used in the text later on, these and _local_file_attrs should be "## Per package tunables" instead I think.

Reply:
Also makes me wonder if the split to tunables really makes sense in the synopsis for fileattr and generators, because eg flags is just as fundamental part of the matching as the path and the magic are, only _opts is a proper tunable, and then having a separate section just for one item... :smile: 

The package-level things are the real tunables here, and _opts belongs there, really.